addons/hr_attendance/models/ir_http.py

```diff
@@ -0,0 +1,15 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.http import request
+
+class Http(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @api.model
+    def get_frontend_session_info_hr_attendance(self):
+        frontend_session_info = super().get_frontend_session_info()
+        frontend_session_info.update({
+            'user_context': request.env.context
```

Comment:
`open_kiosk_mode` in `controllers/main.py`

Reply:
this will be the first time we leak the entire user context to the frontend, I would prefer that you only expose the few specific context value you need rather than the entire context

Reply:
Actually this method is only used by one controller, I think it would be better to move it to that controller. My first impression was that you were overriding the generic `get_frontend_session_info`, not creating a new one. So we are actually not leaking anything to the frontend